### ISSUE TITLE
Can't remove redundant list item on a new line when using List plugin together with Indent plugin

### ISSUE THREAD

**Issue by bahung1221** [ckeditor/ckeditor5#8543]
Hi there 😄 

## 📝 Provide detailed reproduction steps (if any)
Like the title, I can't remove redundant list item on a new line, that was auto-created when press enter. 
This error only occured when using `List plugin` together with `Indent plugin`
I found that I can remove it by press **list icon** on the toolbar, but for me it just a workaround and seems unconventional for UX, because the normal user will **press back button**/**press enter again** to remove list instead press **list icon**

![Screen-Recording-2020-11-28-at-1](https://user-images.githubusercontent.com/27690861/100512188-69d51500-31a1-11eb-80f6-637f3a3c55a2.gif)

Best regards

### ✔️ Expected result
Able to remove redundant list item

### ❌ Actual result
Can't remove redundant list item

## 📃 Other details

* Browser: Chrome
* OS: Macos
* CKEditor version: 23.1.0

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by LukaszGudel** [comment]
Hello, I have looked into your issue and if I understand your use case correctly, then it is working as intended and there is no bug. At the beginning, you have a list with one long list item. This list item contains soft line brakes (\<br/> made with Shift+Enter shortcut). After pressing Enter with caret next to line break it seems like editor creates empty list item, that would disappear when pressing Enter once more. In fact, this list item is not empty and it contains break line tag along with all the rest of the text. Because of this, when you press Enter once more then it creates another list item - this time empty one. Let me know if this helps or is there something more about this issue that I overlooked.
